drv/lpc55-swd/src/main.rs

```diff
@@ -21,34 +21,42 @@
 // A successful read operation consists of three phases:
 // - an eight-bit read packet request, from the host to the target
 // - a three-bit OK acknowledge response, from the target to the host
-// - a 33-bit data read phase, where data is transferred from the target to the host
+// - a 33-bit data read phase, where data is transferred from the target to the
+//   host
 //
 // There is one bit of tunraound between the request and acknowledge.
 //
-// It turns out this specification can be implemented on top of a SPI block with
-// some specific implementation details:
+// It turns out this specification can be implemented on top of a SPI block
+// with some specific implementation details:
+//
 // - SPI has 4 pins (MOSI, MISO, CS, CSK) and importantly expects MOSI and MISO
-// to be separate. This is in contrast to SWD which assumes a single pin for
-// both input and output. For our SPI implementation, we tie MOSI and MISO
-// together and configure exactly one of MOSI or MISO for use with SPI at a time.
-// A side effect of this choice is that reading needs to be precise with the
-// specification. Adding extra idle cycles is fairly easy when writing but that
-// is not possible with reading.
+//   to be separate. This is in contrast to SWD which assumes a single pin for
+//   both input and output. For our SPI implementation, we tie MOSI and MISO
+//   together and configure exactly one of MOSI or MISO for use with SPI at a
+//   time.
+//
+//   A side effect of this choice is that reading needs to be precise with the
+//   specification. Adding extra idle cycles is fairly easy when writing but
+//   that is not possible with reading.
+//
 // - The LPC55 can transmit between 4 and 16 bits at a time. This makes
-// makes transmitting the various combinations of phases and turnaround bits
-// a bit of a pain. This is broken up into the following combinations:
-// -- 8-bit packet write
-// -- 4-bit ACK read (turnaround + three bits of response)
-// -- 34-bit write (one bit turnaround, 32 bits data, one bit partiy) broken up
-//    into 9 bit + 8 bit + 8 bit + 9 bit writes
-// -- 33-bit read (32 bits data, one bit parity) broken up into
-//    8 bit + 8 bit + 8 bit + 9 bit reads. There is also one bit of turnaround
-//    after the read but this is aborbed into idle cycles.
+//   makes transmitting the various combinations of phases and turnaround bits
+//   a bit of a pain. This is broken up into the following combinations:
+//
+//   -- 8-bit packet write
+//   -- 4-bit ACK read (turnaround + three bits of response)
+//   -- 34-bit write (one bit turnaround, 32 bits data, one bit partiy) broken
```

Comment:
s/partiy/parity

Reply:
no, you see, it is a one-bit _party_ :partying_face:

Reply:
Fixed!